pos_analytic_by_config/models/pos_session.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2021 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import models
+
+
+class PosSession(models.Model):
+    _inherit = "pos.session"
+
+    def _credit_amounts(
+        self,
+        partial_move_line_vals,
+        amount,
+        amount_converted,
+        force_company_currency=False,
+    ):
+        """This is where the actual `account.move.line` is created. We can catch here
+        the context coming grom `_get_sale_vals`
+        """
+        account_analytic_id = self.env.context.get("account_analytic_id")
```

Comment:
Why do you need to pass it by context instead of evaluating here `self.config_id.account_analytic_id`?

Reply:
We only want the analyitic account set in the sales items from the account move. This is called from `_get_sale_vals` like this: https://github.com/odoo/odoo/blob/13.0/addons/point_of_sale/models/pos_session.py#L746 but this method is called with other credit operations like https://github.com/odoo/odoo/blob/8b521fdf44b89de1204853dc4bf2f9bb7daca334/addons/point_of_sale/models/pos_session.py#L767

So since we're already checking the the config's analytic account the moment before and flagging the context, why not passing the analytic account id already?

Reply:
OK, you can put this as comment in the code for avoiding the same question in the future.

Reply:
Done